test/hooks-order.js

```diff
@@ -0,0 +1,390 @@
+
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/hooks-order';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+const errors = [{ruleId: 'no-todo-test'}];
+const header = 'const test = require(\'ava\');\n';
+
+ruleTester.run('no-todo-test', rule, {
+	valid: [
+		`
+			${header}
+
+			test.before((t) => {
+				doFoo();
+			});
+
+			test.after((t) => {
+				doFoo();
+			});
+
+			test.after.always((t) => {
+				doFoo();
+			});
+
+			test.beforeEach((t) => {
+				doFoo();
+			});
+
+			test.afterEach((t) => {
+				doFoo();
+			});
+
+			test.afterEach.always((t) => {
+				doFoo();
+			});
+
+			test('foo', (t) => {
+				t.true(true);
+			});
+		`,
+		`
+			test.before((t) => {
```

Comment:
The point of "headless" tests is to validate that the rule don't report an error on non-test files. It should be an invalid case, not reported because out of a test file.

Reply:
I don't understand what you mean. Do you want me to move it to the `invalid` section but with no `errors`?

I assumed the `valid` / `invalid` areas of the test file were named that because of how the test worked.

Reply:
I'm sorry if it wasn't clear, I mean that rules should not report errors outside of test file. Those test files are detected by the presence of `import test from 'ava';`.

A seemingly invalid test-case without this header should not fail and therefore be valid. (e.g [`max-asserts`](https://github.com/avajs/eslint-plugin-ava/blob/master/test/max-asserts.js#L40))

Reply:
Isn't that what this is doing? The case you are commenting on will not report a lint error for the exact reason you state.

Reply:
Yep, however the point is to write an invalid test-case to check that it didn't report because it miss a "header" and not because it's a valid test-case.

This is valid:
```js
// No header, wrong order
test.after(t => t.true(1 === 1));

test.before(t => t.true(1 === 1));
```

Reply:
Ah, gotcha. That's exact case is on [line 88](https://github.com/avajs/eslint-plugin-ava/pull/265/files/510784dc44c71189691d05f8d34dd8eb0958d322#diff-24bb544a83a06a413ab2ad3d665bfdcaR88):

```
		`
 			test.after((t) => {
 				doFoo();
 			});

 			test.before((t) => {
 				doFoo();
 			});
 		`,
```